ambry-commons/src/main/java/com.github.ambry.commons/ReadableStreamChannelInputStream.java

```diff
@@ -107,7 +107,14 @@ private boolean loadData()
         asyncWritableChannel.resolveOldestChunk(null);
       }
       try {
-        currentChunk = asyncWritableChannel.getNextChunk();
+        while (true) {
```

Comment:
I think this method should throw the `InterruptedException` and let the caller deal with it.
Also, are you sure this method cannot be made simpler? I find certain things repetitive in here.